projects/vdk-core/src/vdk/internal/core/config.py

```diff
@@ -65,11 +64,25 @@ class Configuration:
         default_factory=dict
     )
     __config_key_to_sensitive: dict[ConfigKey, bool] = field(default_factory=dict)
+    __config_key_to_can_override: dict[ConfigKey, bool] = field(default_factory=dict)
 
     def __getitem__(self, key: ConfigKey):
         key = _normalize_config_key(key)
         return self.get_value(key)
 
+    def __can_override(self, key: ConfigKey):
+        return self.__config_key_to_can_override.get(key, False)
+
+    def override_with_secret(self, key: ConfigKey, value: ConfigValue):
```

Comment:
```python
newbuilder = ConfigurationBuilder(current_configuration)
newbuilder.set_value ...
context.config = newbuilder.get()   
```

I tried this and it doesn't work, because the context class is also a frozen dataclass. I went with your suggestion to rename the method. Also removed the locking. Seems a bit pointless.